### PR TITLE
[Refactor] Improved Logging for IdentityServer

### PR DESCRIPTION
**Pull request by komer3**
- Add comments for IdentityServer struct and methods
- Implement structured logging using logger.GetLogger
- Remove dependency on k8s.io/klog/v2
- Add log entries for request processing and completion

### General:

* [ ] Have you removed all sensitive information, including but not limited to access keys and passwords?
* [ ] Have you checked to ensure there aren't other open or closed [Pull Requests](../../pulls) for the same bug/feature/question?

### Pull Request Guidelines:

1. [ ] Does your submission pass tests?
1. [ ] Have you added tests? 
1. [ ] Are you addressing a single feature in this PR? 
1. [ ] Are your commits atomic, addressing one change per commit?
1. [ ] Are you following the conventions of the language? 
1. [ ] Have you saved your large formatting changes for a different PR, so we can focus on your work?
1. [ ] Have you explained your rationale for why this feature is needed? 
1. [ ] Have you linked your PR to an [open issue](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)

### CODE REVIEW

File: internal/driver/identityserver.go [hunk 1]
```diff
@@ -50,22 +51,36 @@ func NewIdentityServer(ctx context.Context, linodeDriver *LinodeDriver) (*Identi
...
+	log, _, done := logger.GetLogger(ctx).WithMethod("GetPluginInfo")
+	defer done()
+
+	log.V(2).Info("Processing request")
 
 	if linodeIdentity.driver.name == "" {
 		return nil, status.Error(codes.Unavailable, "Driver name not configured")
 	}
 
+	log.V(2).Info("Successfully completed")
```
Comment (amold1) [inline internal/driver/identityserver.go:67]: logging on line 61 and 67 doesnt seem right to me... we havent really processed the request yet... it happens when we return with no error on line 68-71.
I would suggest removing these and adding those logs before and after the function/method that calls `GetPluginInfo`

File: internal/driver/identityserver.go [hunk 2]
```diff
@@ -50,22 +51,36 @@ func NewIdentityServer(ctx context.Context, linodeDriver *LinodeDriver) (*Identi
...
+// This method is REQUIRED for the Identity service as per the CSI spec.
+// It informs the CO of the supported features by this plugin.
 func (linodeIdentity *IdentityServer) GetPluginCapabilities(ctx context.Context, req *csi.GetPluginCapabilitiesRequest) (*csi.GetPluginCapabilitiesResponse, error) {
-	klog.V(5).Infof("Using default GetPluginCapabilities")
+	log, _, done := logger.GetLogger(ctx).WithMethod("GetPluginCapabilities")
+	defer done()
+
+	log.V(2).Info("Processing request")
+
+	log.V(2).Info("Successfully completed")
```
Comment (amold1) [inline internal/driver/identityserver.go:83]: same comment as above

File: internal/driver/identityserver.go [hunk 3]
```diff
@@ -98,11 +113,19 @@ func (linodeIdentity *IdentityServer) GetPluginCapabilities(ctx context.Context,
...
-	klog.V(4).Infof("Probe called with args: %#v", req)
+	log, _, done := logger.GetLogger(ctx).WithMethod("Probe")
+	defer done()
+
+	log.V(2).Info("Processing request")
+
 	linodeIdentity.driver.readyMu.Lock()
 	defer linodeIdentity.driver.readyMu.Unlock()
 
+	log.V(2).Info("Successfully completed")
```
Comment (amold1) [inline internal/driver/identityserver.go:128]: and here

File: internal/driver/identityserver.go (hunk 1 above)
Comment (komer3) [reply internal/driver/identityserver.go]: Removed it. I had that there to sort of capture entry and exit for the func. But these are super simple grpc funcs() so we should be fine just removing `Successfully completed `

**Review by amold1 — APPROVED** [review]

**Review by shanduur — DISMISSED** [review]

**Review by amold1 — APPROVED** [review]

**Review by eljohnson92 — APPROVED** [review]

**Pull request merged**